src/extract/multipart.rs

```diff
@@ -80,6 +81,21 @@ impl Multipart {
             Ok(None)
         }
     }
+
+    /// make all fields to a hashmap
+    /// you can easy to check & get field data.
+    pub async fn map(&mut self) -> Result<HashMap<String, Field<'_>>, MultipartError> {
+
+        let mut result = HashMap::new();
+
+        while let Some(mut field) = self.next_field().await? {
+            let name = field.name()?.to_string();
+            result.insert(name, field);
+        }
+
+        Ok(result)
+    }
```

Comment:
Did you actually compile this? I don't think it compiles, or at least it _shouldn't_.

There can only be one `Field` live at any point, since `Field` contains `&mut Multipart`. Multiple `Field`s from the same `Multipart` would mean there are multiple `&mut Multipart` which is unsound. Here you're storing multiple `Field`s in the map at the same time. This is multers ["Field exclusivity"](https://docs.rs/multer/2.0.1/multer/struct.Multipart.html#field-exclusivity) enforced via lifetimes.

Reply:
Sorry, I'll try to fix it. I just made a brief proposal, and I wonder if you have a problem with him.